src/Service/BlogService.php

```diff
@@ -5,10 +5,12 @@
 
 class BlogService
 {
-    private static function getClient()
+    private $client;
+
+    public function __construct($blogUrl, $username, $password)
     {
-        $wp = new WordpressClient($_ENV['WP_ENDPOINT'], $_ENV['WP_USER'], $_ENV['WP_PASSWORD']);
-        return $wp;
+        $url = rtrim($blogUrl, "/") . "/xmlrpc.php";
+        $this->client = new WordpressClient($url, $username, $password);
```

Comment:
This isn't really what I meant. The `BlogService` should consume a fully configured `WordpressClient` instead of creating one. Any time you have `new Foo`, it means you're bypassing dependency injection and breaking dependency inversion.